Detectors/HMPID/calibration/testWorkflow/HMPIDDCSDataProcessorSpec.h

```diff
@@ -0,0 +1,229 @@
+      std::string ccdbpath = ic.options().get<std::string>("ccdb-path");
+      auto& mgr = CcdbManager::instance();
+      mgr.setURL(ccdbpath);
+      CcdbApi api;
+      api.init(mgr.getURL());
+      long ts = std::chrono::duration_cast<std::chrono::milliseconds>(std::chrono::system_clock::now().time_since_epoch()).count();
+      std::unordered_map<DPID, std::string>* dpid2DataDesc = mgr.getForTimeStamp<std::unordered_map<DPID, std::string>>("HMP/Config/DCSDPconfig", ts);
+      for (auto& i : *dpid2DataDesc) {
+        vect.push_back(i.first);
+      }
+    } else {
+      LOG(info) << "Configuring via hardcoded strings";
+      std::vector<std::string> expaliases = o2::dcs::expandAliases(aliases);
+
+      for (const auto& i : expaliases) {
+        vect.emplace_back(i, o2::dcs::DPVAL_DOUBLE);
+        // LOG(info) << i;
+      }
+
+    } // end else
+
+    // LOG(info) << "Listing Data Points for HMPID:";
+    // for (auto& i : vect) {
+    // LOG(info) << i;
+    // }
+
+    mProcessor = std::make_unique<o2::hmpid::HMPIDDCSProcessor>();
+    bool useVerboseMode = ic.options().get<bool>("use-verbose-mode");
+    LOG(info) << " ************************* Verbose?" << useVerboseMode;
+
+    if (useVerboseMode) {
+      mProcessor->useVerboseMode();
+    }
+    mProcessor->init(vect);
+    mTimer = HighResClock::now();
+  }
+
+  //==========================================================================
+
+  void run(o2::framework::ProcessingContext& pc) final
+  {
+    if (mCheckRunStartStop) {
+      const auto* grp = mRunChecker.check(); // check if there is a run with HMP
+      // this is an example of what it will return
+      if (mRunChecker.getRunStatus() == RunStatus::NONE) {
+        LOGP(info, "No run with is ongoing or finished");
+      } else if (mRunChecker.getRunStatus() == RunStatus::START) { // saw new run with wanted detectors
+        LOGP(info, "Run {} has started", mRunChecker.getFollowedRun());
+        grp->print();
+      } else if (mRunChecker.getRunStatus() == RunStatus::ONGOING) { // run which was already seen is still ongoing
+        LOGP(info, "Run {} is still ongoing", mRunChecker.getFollowedRun());
+      } else if (mRunChecker.getRunStatus() == RunStatus::STOP) { // run which was already seen was stopped (EOR seen)
+        LOGP(info, "Run {} was stopped", mRunChecker.getFollowedRun());
+      }
+    }
+
+    auto startValidity = DataRefUtils::getHeader<DataProcessingHeader*>(pc.inputs().getFirstValid(true))->creation;
+
+    auto timeNow = HighResClock::now();
+
+    // process datapoints:
+    auto dps = pc.inputs().get<gsl::span<DPCOM>>("input");
+    mProcessor->process(dps);
+    if (startValidity == 0xffffffffffffffff) {                                                                   // it means it is not set
+      startValidity = std::chrono::duration_cast<std::chrono::milliseconds>(timeNow.time_since_epoch()).count(); // in ms
+    }
+
+    mProcessor->setStartValidity(startValidity);
```

Comment:
you are setting startvalidity each time you receive something, i.e. at least every 5 s (when the full DPs image is sent even if nothing was changed). Should not you use as a start validity the beginning of the integration period?